acceptance/cluster/localcluster.go

```diff
@@ -504,11 +504,18 @@ func (l *LocalCluster) Start() {
 func (l *LocalCluster) Assert(t *testing.T) {
 	const almostZero = 50 * time.Millisecond
 	filter := func(ch chan Event, wait time.Duration) *Event {
+		waitTimer := time.NewTimer(wait)
+		waitTimerRead := false
 		for {
+			if !waitTimer.Reset(wait) && !waitTimerRead {
+				<-waitTimer.C
+			}
+			waitTimerRead = false
 			select {
 			case act := <-ch:
 				return &act
-			case <-time.After(wait):
+			case <-waitTimer.C:
+				waitTimerRead = true
```

Comment:
I've added a new `util.Timer` type that does exactly what you were describing in your first comment. It's unfortunate to have to wrap the standard library, but I agree it's better than having bug workarounds throughout our code. I've also added deferred `Timer.Stop` calls where needed. PTAL